API.adoc

```diff
@@ -0,0 +1,402 @@
+URNs or identifiers are just `String` values in the scope of REST modules. The
+scheme must of course be compatible with the underlying data store.
+
+=== Request Parameters
+[width="100%",options="header"]
+|====================
+| Parameter | Parameter Type | Format | Default | Description
+| `type` | url | String | no default | the type of the Thing (e.g., "Employee", "Building")
+| `urn` | url | String | no default | the URN of the Thing
+| `fields` | query | Comma-separated String list  |  query parameter, not set by default | Metadata to include in the response (defaults to all if not set)
+| `page` | query | String representation of Long  | 1 | page number of multi-element paged returns, starting at 1
+| `size` | query | String representation of Long  | 20 | number of elements per page in multi-element paged responses
+| `sortOrder` | query | String of {"asc", "desc"}  | "asc" | sort order for paged responses
+| `sortBy` | query | String | chronological order | paged responses sorted by this JSON field name
+|====================
+
+
+=== API Endpoints
+
+==== Common Error Responses
+
+:grid: rows
+[frame="topbot", cols="2,4", options="header"]
+|====
+| Response| Description
+| 400 BAD REQUEST | There were constraint violations in the request body.
+| 401 UNAUTHORIZED | The User represented by the authentication header could not be authenticated.
+| 403 FORBIDDEN | The User represented by the authentication header lacks the authority to perform this action.
+| 404 NOT FOUND | The Thing or Metadata was not found.
+| 409 CONFLICT | A Thing with this URN already exists.
+|====
+:grid: all
+
+[[create1]]
+==== Create - `POST /{type}`
+
+----
+POST /Building
+----
+
+Create a new Thing belonging to the Tenant of the authenticated User, and its associated
+Metadata. In the first example, the fields `name`, `description` and `large` will be
+added as Metadata for this Thing.
+
+*Example 1*
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
+    "active": true,
+    "name": "My home",
+    "description": "Where I live right now",
+    "large": true
+}
+----
+
+.Response
+----
+201 CREATED
+----
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
+    "type": "Building",
+    "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+    "active": true
+}
+----
+
+*Example 2*
+[source,json]
+----
+{
+}
+----
+
+.Response
+----
+201 CREATED
+----
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:2519a8ba-fadf-4a85-a965-5a59a5b43e7d",
+    "type": "Building",
+    "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+    "active": true
+}
+----
+
+[[update1]]
+==== Update - `PUT /{type}/{urn}`
+
+Update an already existing Thing belonging to the Tenant of the authenticated User,
+and update its associated Metadata.
+
+NOTE: Type, URN, and Tenant URN are immutable for an existing Thing; this endpoint
```

Comment:
`PUT` also updates the Metadata entry. Thus, I'd remove "this endpoint is only useful for turning the active flag off and on"

Instead, we could put something like "This endpoint is useful to turn the active flag off and on, and to update Metadata"